src/main/java/io/debezium/connector/cassandra/SchemaProcessor.java

```diff
@@ -139,34 +141,45 @@ public void onTableRemoved(final TableMetadata table) {
                     final String ksName = table.getKeyspace().getName();
                     final String tableName = table.getName();
 
-                    final org.apache.cassandra.schema.KeyspaceMetadata oldKsm = Schema.instance.getKSMetaData(table.getKeyspace().getName());
+                    final org.apache.cassandra.schema.KeyspaceMetadata oldKsm = Schema.instance.getKSMetaData(ksName);
 
                     if (oldKsm == null) {
-                        LOGGER.warn("KeyspaceMetadata for keyspace {} is not found!", table.getKeyspace().getName());
+                        LOGGER.warn("KeyspaceMetadata for keyspace {} is not found!", ksName);
                         return;
                     }
 
-                    final ColumnFamilyStore cfs = Keyspace.openWithoutSSTables(ksName).getColumnFamilyStore(tableName);
+                    ColumnFamilyStore cfs;
 
-                    if (cfs == null) {
-                        LOGGER.warn("ColumnFamilyStore for {}.{} is not found!", table.getKeyspace(), table.getName());
+                    try {
+                        cfs = Keyspace.openWithoutSSTables(ksName).getColumnFamilyStore(tableName);
+                    }
+                    catch (final Exception ex) {
+                        LOGGER.warn("ColumnFamilyStore for {}.{} is not found!", ksName, table.getName());
                         return;
                     }
 
                     // make sure all the indexes are dropped, or else.
                     cfs.indexManager.markAllIndexesRemoved();
 
                     // reinitialize the keyspace.
-                    final CFMetaData cfm = oldKsm.tables.get(tableName).get();
-                    final org.apache.cassandra.schema.KeyspaceMetadata newKsm = oldKsm.withSwapped(oldKsm.tables.without(tableName));
+                    final Optional<CFMetaData> cfm = oldKsm.tables.get(tableName);
 
-                    Schema.instance.unload(cfm);
-                    Schema.instance.setKeyspaceMetadata(newKsm);
+                    final Method unregisterMBeanMethod = ColumnFamilyStore.class.getDeclaredMethod("unregisterMBean");
+                    unregisterMBeanMethod.setAccessible(true);
+                    unregisterMBeanMethod.invoke(cfs);
```

Comment:
So that'll be quite troublesome when moving to newer Java versions (once supported by Cassandra). What's the reasoning for doing this reflectively? A comment in the code would be warranted at the very least, and we'll have to look for alternatives going forward.

Reply:
Why would moving to newer Java cause a problem here? Btw afaik Cassandra supports 8 and 11 now only. (and 11 only in case of Cassandra 4 this connector in incompatible anyway yet).

Reply:
In newer version, reflection is more and more frowned upon. Java 16 by default disallows reflection on JDK internal types, the same when trying to do this with modules (whenever that'd be the case here). So it's not an immediate problem, but the more we can avoid this sort of reflective access, the better. So why is it this method needs to be invoked that way?

Reply:
That method is package protected so we would have to accommodate the code to `org.apache.cassandra.db`. I mean, be might indeed do that if you dont mind shuffling with the code here and there.

The way how it would look like is like this:

https://github.com/debezium/debezium-connector-cassandra/pull/33

Reply:
Replied over there; thx.